I suppose this may be related…
```
2024-06-07T15:26:06.5371531Z [ERROR] io.quarkus.vertx.deployment.MessageConsumerFailureTest.testFailure
2024-06-07T15:26:06.5373087Z [ERROR]   Run 1: MessageConsumerFailureTest.testFailure:49->verifyFailure:110 expected: <boom> but was: <java.io.IOException: boom>
2024-06-07T15:26:06.5374337Z [ERROR]   Run 2: MessageConsumerFailureTest.testFailure:49->verifyFailure:110 expected: <boom> but was: <java.io.IOException: boom>
2024-06-07T15:26:06.5375516Z [ERROR]   Run 3: MessageConsumerFailureTest.testFailure:49->verifyFailure:110 expected: <boom> but was: <java.io.IOException: boom>
2024-06-07T15:26:06.5376704Z [ERROR]   Run 4: MessageConsumerFailureTest.testFailure:49->verifyFailure:110 expected: <boom> but was: <java.io.IOException: boom>
```

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -58,7 +58,7 @@
         <smallrye-graphql.version>2.8.4</smallrye-graphql.version>
         <smallrye-fault-tolerance.version>6.3.0</smallrye-fault-tolerance.version>
         <smallrye-jwt.version>4.5.2</smallrye-jwt.version>
-        <smallrye-context-propagation.version>2.1.0</smallrye-context-propagation.version>
+        <smallrye-context-propagation.version>2.1.2</smallrye-context-propagation.version>
         <smallrye-reactive-streams-operators.version>1.0.13</smallrye-reactive-streams-operators.version>
         <smallrye-reactive-types-converter.version>3.0.1</smallrye-reactive-types-converter.version>
         <smallrye-mutiny-vertx-binding.version>3.12.0</smallrye-mutiny-vertx-binding.version>

--- a/extensions/vertx/deployment/src/test/java/io/quarkus/vertx/deployment/MessageConsumerFailureTest.java
+++ b/extensions/vertx/deployment/src/test/java/io/quarkus/vertx/deployment/MessageConsumerFailureTest.java
@@ -46,14 +46,14 @@ public class MessageConsumerFailureTest {
         verifyFailure("foo-completion-stage", "java.lang.NullPointerException: Something is null", false);
         verifyFailure("foo-completion-stage-failure", "boom", true);
         verifyFailure("foo-uni", "java.lang.NullPointerException: Something is null", false);
-        verifyFailure("foo-uni-failure", "boom", true);
+        verifyFailure("foo-uni-failure", "java.io.IOException: boom", true);
 
         verifyFailure("foo-blocking", "java.lang.IllegalStateException: Red is dead", false);
         verifyFailure("foo-message-blocking", "java.lang.NullPointerException", false);
         verifyFailure("foo-completion-stage-blocking", "java.lang.NullPointerException: Something is null", false);
         verifyFailure("foo-completion-stage-failure-blocking", "boom", true);
         verifyFailure("foo-uni-blocking", "java.lang.NullPointerException: Something is null", false);
-        verifyFailure("foo-uni-failure-blocking", "boom", true);
+        verifyFailure("foo-uni-failure-blocking", "java.io.IOException: boom", true);
     }
 
     @Test

--- a/integration-tests/smallrye-context-propagation/src/test/java/io/quarkus/context/test/mutiny/MutinyContextEndpoint.java
+++ b/integration-tests/smallrye-context-propagation/src/test/java/io/quarkus/context/test/mutiny/MutinyContextEndpoint.java
@@ -1,6 +1,7 @@
 package io.quarkus.context.test.mutiny;
 
 import java.net.MalformedURLException;
+import java.time.Duration;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
@@ -460,5 +461,22 @@ public class MutinyContextEndpoint {
         // now delete both entities
         Assertions.assertEquals(2, Person.deleteAll());
         return ret;
+    }
+
+    @GET
+    @Path("/bug40852")
+    public String bug40852() {
+        var futureW = Uni
+                .createFrom()
+                .item("item")
+                .onItem()
+                .delayIt()
+                .by(Duration.ofMillis(100))
+                .subscribeAsCompletionStage();
+
+        futureW.whenComplete((result, error) -> {
+            Assertions.assertEquals(true, futureW.isDone());
+        }).join();
+        return "OK";
     }
 }

--- a/integration-tests/smallrye-context-propagation/src/test/java/io/quarkus/context/test/mutiny/MutinyContextPropagationTest.java
+++ b/integration-tests/smallrye-context-propagation/src/test/java/io/quarkus/context/test/mutiny/MutinyContextPropagationTest.java
@@ -156,6 +156,12 @@ public class MutinyContextPropagationTest {
                 .statusCode(Response.Status.OK.getStatusCode()));
     }
 
+    @Test
+    public void testContextPropagationBug40852() {
+        RestAssured.when().get("/mutiny-context/bug40852").then()
+                .statusCode(Response.Status.OK.getStatusCode());
+    }
+
     private void awaitState(ThrowingRunnable task) {
         Awaitility.await().atMost(5, TimeUnit.SECONDS)
                 .pollInterval(100, TimeUnit.MILLISECONDS)

